src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/SetAzureVMOSDiskCommand.cs

```diff
@@ -48,7 +48,7 @@ public class SetAzureVMOSDiskCommand : Microsoft.Azure.Commands.ResourceManager.
         [Parameter(
             Mandatory = false,
             Position = 1,
-            ValueFromPipelineByPropertyName = true,
+            ValueFromPipelineByPropertyName = false,
```

Comment:
why remove this?

Reply:
When VM object is piped to Set-AzureRmVMOSDisk. VM.name property sets this value.  However, this parameter takes a OSDisk name, not VM name.  Thus, piping scenario breaks if this is set to true.   Changing it to false fixes this issue.